task_02/alexey_mesheryakov/graph.cpp

```diff
@@ -0,0 +1,91 @@
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <utility>
+
+const int LEN = 14;
+
+using std::pair;
+using std::vector;
+using std::cout;
+
+class Graph {
+public:
+	struct Edge{
+		int self_id = 0;
+		pair <int, int> vert = {-1,-1};
+	};
+	
+	struct Vert{
+		int inf;
+		vector<int> edges = {};
+	};
+
+	void set_vert_inf(int id, int inf) {
+		vert_inf[id].inf = inf;
+	}
+	
+	void set_vert(Vert v) {
+		vert_inf.push_back(v);  
+	}
+
+	void set_vert_inf(vector<Vert> inf) {
+		vert_inf = inf;
+	}
+
+	void set_edge(Edge edge) {
+	 	edges.push_back(edge);
+		vert_inf[edge.vert.first].edges.push_back(edge.self_id);
+		vert_inf[edge.vert.second].edges.push_back(edge.self_id);
+	}
+
+	Graph (vector<Edge> newedges, vector<int> inf) {
+		for (auto v: inf)
+			set_vert({v,{}}); 
+		for (auto edge: newedges)
+			set_edge(edge);
+	}
+
+	void print() {
+		for (auto edge: edges)
+			cout<< edge.self_id << " " << vert_inf[edge.vert.first].inf << " "<< vert_inf[edge.vert.second].inf <<"\n";
+	}
+
+	void printjson() {
```

Comment:
Опять же, работа с файлами не входит в зону отвественности графа.
Эта функция должна возвращать строку, которая извне уже может быть записана в файл.
```suggestion
      std::string to_json() const {
```